Typescript imports referring to .js files instead of .ts

**Describe the bug**
This might be more a question than a bug, but I'm working on a Typescript/Webpack 5 project and replacing Selectize with Tom Select.

Since I'm compiling my own Typescript down I decided to import straight from the source files:
```
import TomSelect from 'tom-select/src/tom-select';
//import plugins, etc.
```

All of the relative import statements are using the .js extension which don't exist (since they're all .ts)
`import TomSelect from '../../tom-select.js';`

Is there a reason the extension is JS or can it easily be changed to TS? Or is there some tsconfig setting to load the TS files instead of the referenced JS files?

**To Reproduce**
Create a typescript file and import TomSelect:
`import TomSelect from 'tom-select/src/tom-select';`

Error:
```
ERROR in ./node_modules/tom-select/src/plugins/checkbox_options/plugin.ts 15:0-44
Module not found: Error: Can't resolve '../../tom-select.js' in '{...}\node_modules\tom-select\src\plugins\checkbox_options'
```

**Expected behavior**
I should be able to refer to the Typescript files and my webpack ts-loader handle them.

Context:

- tom-select: 2.0.0-rc.3
- webpack: 5.52.0
- typescript: 4.3.4

--- a/src/getSettings.ts
+++ b/src/getSettings.ts
@@ -1,4 +1,4 @@
-import defaults from './defaults.js';
+import defaults from './defaults';
 import { hash_key } from './utils';
 import { TomOption, TomSettings } from './types/index';
 import { iterate } from '@orchidjs/sifter/lib/utils';

--- a/src/plugins/caret_position/plugin.ts
+++ b/src/plugins/caret_position/plugin.ts
@@ -13,7 +13,7 @@
  *
  */
 
-import TomSelect from '../../tom-select.js';
+import TomSelect from '../../tom-select';
 import { nodeIndex } from '../../vanilla';
 
 

--- a/src/plugins/change_listener/plugin.ts
+++ b/src/plugins/change_listener/plugin.ts
@@ -13,7 +13,7 @@
  *
  */
 
-import TomSelect from '../../tom-select.js';
+import TomSelect from '../../tom-select';
 import { addEvent } from '../../utils';
 
 TomSelect.define('change_listener',function(this:TomSelect){

--- a/src/plugins/checkbox_options/plugin.ts
+++ b/src/plugins/checkbox_options/plugin.ts
@@ -12,7 +12,7 @@
  * governing permissions and limitations under the License.
  *
  */
-import TomSelect from '../../tom-select.js';
+import TomSelect from '../../tom-select';
 import { preventDefault, hash_key } from '../../utils';
 import { getDom } from '../../vanilla';
 

--- a/src/plugins/clear_button/plugin.ts
+++ b/src/plugins/clear_button/plugin.ts
@@ -13,7 +13,7 @@
  *
  */
 
-import TomSelect from '../../tom-select.js';
+import TomSelect from '../../tom-select';
 import { getDom } from '../../vanilla';
 import { CBOptions } from './types';
 

--- a/src/plugins/drag_drop/plugin.ts
+++ b/src/plugins/drag_drop/plugin.ts
@@ -12,7 +12,7 @@
  * governing permissions and limitations under the License.
  *
  */
-import TomSelect from '../../tom-select.js';
+import TomSelect from '../../tom-select';
 
 TomSelect.define('drag_drop',function(this:TomSelect) {
 	var self = this;

--- a/src/plugins/dropdown_header/plugin.ts
+++ b/src/plugins/dropdown_header/plugin.ts
@@ -13,7 +13,7 @@
  *
  */
 
-import TomSelect from '../../tom-select.js';
+import TomSelect from '../../tom-select';
 import { getDom } from '../../vanilla';
 import { preventDefault } from '../../utils';
 import { DHOptions } from './types';

--- a/src/plugins/dropdown_input/plugin.ts
+++ b/src/plugins/dropdown_input/plugin.ts
@@ -13,8 +13,8 @@
  *
  */
 
-import TomSelect from '../../tom-select.js';
-import * as constants from '../../constants.js';
+import TomSelect from '../../tom-select';
+import * as constants from '../../constants';
 import { getDom, addClasses } from '../../vanilla';
 import { addEvent, preventDefault } from '../../utils';
 

--- a/src/plugins/input_autogrow/plugin.ts
+++ b/src/plugins/input_autogrow/plugin.ts
@@ -11,7 +11,7 @@
  * governing permissions and limitations under the License.
  *
  */
-import TomSelect from '../../tom-select.js';
+import TomSelect from '../../tom-select';
 import { addEvent } from '../../utils';
 
 TomSelect.define('input_autogrow', function(this:TomSelect) {

--- a/src/plugins/no_active_items/plugin.ts
+++ b/src/plugins/no_active_items/plugin.ts
@@ -11,7 +11,7 @@
  * governing permissions and limitations under the License.
  *
  */
-import TomSelect from '../../tom-select.js';
+import TomSelect from '../../tom-select';
 
 TomSelect.define('no_active_items', function(this:TomSelect) {
 	this.hook('instead','setActiveItem',() => {});

--- a/src/plugins/no_backspace_delete/plugin.ts
+++ b/src/plugins/no_backspace_delete/plugin.ts
@@ -11,7 +11,7 @@
  * governing permissions and limitations under the License.
  *
  */
-import TomSelect from '../../tom-select.js';
+import TomSelect from '../../tom-select';
 
 TomSelect.define('no_backspace_delete', function(this:TomSelect) {
 	var self = this;

--- a/src/plugins/optgroup_columns/plugin.ts
+++ b/src/plugins/optgroup_columns/plugin.ts
@@ -13,8 +13,8 @@
  *
  */
 
-import TomSelect from '../../tom-select.js';
-import * as constants from '../../constants.js';
+import TomSelect from '../../tom-select';
+import * as constants from '../../constants';
 import { parentMatch, nodeIndex } from '../../vanilla';
 
 TomSelect.define('optgroup_columns', function(this:TomSelect) {

--- a/src/plugins/remove_button/plugin.ts
+++ b/src/plugins/remove_button/plugin.ts
@@ -12,7 +12,7 @@
  * governing permissions and limitations under the License.
  *
  */
-import TomSelect from '../../tom-select.js';
+import TomSelect from '../../tom-select';
 import { getDom } from '../../vanilla';
 import { escape_html, preventDefault, addEvent } from '../../utils';
 import { TomOption } from '../../types/index';

--- a/src/plugins/restore_on_backspace/plugin.ts
+++ b/src/plugins/restore_on_backspace/plugin.ts
@@ -12,7 +12,7 @@
  * governing permissions and limitations under the License.
  *
  */
-import TomSelect from '../../tom-select.js';
+import TomSelect from '../../tom-select';
 import { TomOption } from '../../types/index';
 
 type TPluginOptions = {

--- a/src/plugins/virtual_scroll/plugin.ts
+++ b/src/plugins/virtual_scroll/plugin.ts
@@ -12,7 +12,7 @@
  * governing permissions and limitations under the License.
  *
  */
-import TomSelect from '../../tom-select.js';
+import TomSelect from '../../tom-select';
 import { TomOption } from '../../types/index';
 import { addClasses } from '../../vanilla';
 

--- a/src/tom-select.complete.ts
+++ b/src/tom-select.complete.ts
@@ -1,18 +1,18 @@
 import TomSelect from './tom-select';
-import './plugins/change_listener/plugin.js';
-import './plugins/checkbox_options/plugin.js';
-import './plugins/clear_button/plugin.js';
-import './plugins/drag_drop/plugin.js';
-import './plugins/dropdown_header/plugin.js';
-import './plugins/caret_position/plugin.js';
-import './plugins/dropdown_input/plugin.js';
-import './plugins/input_autogrow/plugin.js';
-import './plugins/no_backspace_delete/plugin.js';
-import './plugins/no_active_items/plugin.js';
-import './plugins/optgroup_columns/plugin.js';
-import './plugins/remove_button/plugin.js';
-import './plugins/restore_on_backspace/plugin.js';
-import './plugins/virtual_scroll/plugin.js';
+import './plugins/change_listener/plugin';
+import './plugins/checkbox_options/plugin';
+import './plugins/clear_button/plugin';
+import './plugins/drag_drop/plugin';
+import './plugins/dropdown_header/plugin';
+import './plugins/caret_position/plugin';
+import './plugins/dropdown_input/plugin';
+import './plugins/input_autogrow/plugin';
+import './plugins/no_backspace_delete/plugin';
+import './plugins/no_active_items/plugin';
+import './plugins/optgroup_columns/plugin';
+import './plugins/remove_button/plugin';
+import './plugins/restore_on_backspace/plugin';
+import './plugins/virtual_scroll/plugin';
 
 
 export default TomSelect;

--- a/src/tom-select.popular.ts
+++ b/src/tom-select.popular.ts
@@ -1,9 +1,9 @@
 import TomSelect from './tom-select';
-import './plugins/caret_position/plugin.js';
-import './plugins/dropdown_input/plugin.js';
-import './plugins/no_backspace_delete/plugin.js';
-import './plugins/remove_button/plugin.js';
-import './plugins/restore_on_backspace/plugin.js';
+import './plugins/caret_position/plugin';
+import './plugins/dropdown_input/plugin';
+import './plugins/no_backspace_delete/plugin';
+import './plugins/remove_button/plugin';
+import './plugins/restore_on_backspace/plugin';
 
 
 export default TomSelect;

--- a/src/tom-select.ts
+++ b/src/tom-select.ts
@@ -1,12 +1,12 @@
 
-import MicroEvent from './contrib/microevent.js';
-import MicroPlugin from './contrib/microplugin.js';
+import MicroEvent from './contrib/microevent';
+import MicroPlugin from './contrib/microplugin';
 import Sifter from '@orchidjs/sifter/lib/sifter';
 import { escape_regex, iterate } from '@orchidjs/sifter/lib/utils';
 import { TomInput, TomArgObject, TomOption, TomOptions, TomCreateFilter, TomCreateCallback, TomItem, TomSettings, TomTemplateNames } from './types/index';
-import {highlight, removeHighlight} from './contrib/highlight.js';
-import * as constants from './constants.js';
-import getSettings from './getSettings.js';
+import {highlight, removeHighlight} from './contrib/highlight';
+import * as constants from './constants';
+import getSettings from './getSettings';
 import {
 	hash_key,
 	get_hash,

